test/Spago/Build.purs

```diff
@@ -161,6 +161,22 @@ spec = Spec.around withTempDir do
       spagoYaml <- FS.readTextFile "spago.yaml"
       spagoYaml `shouldContain` "- prelude: \">=6.0.1 <7.0.0\""
 
+    Spec.it "failed build with many warnings and --json-errors does not truncate output" \{ spago, fixture } -> do
+      spago [ "init" ] >>= shouldBeSuccess
+      let
+        srcMain = Path.concat [ "src", "Main.purs" ]
+        srcWarn = Path.concat [ "src", "Warn.purs" ]
+      FSA.unlink srcMain
+      FS.copyFile
+        { src: fixture "build-json-truncated-main.purs"
+        , dst: srcMain
+        }
+      FS.copyFile
+        { src: fixture "build-json-truncated-warn.purs"
+        , dst: srcWarn
+        }
```

Comment:
If you move the fixtures into their own `build-json-truncated` directory and structure it like a spago project then you can just use `copyTree`, see e.g. the monorepo tests:
https://github.com/purescript/spago/blob/ebb99a02dacd119e8b650855bfae308798c46c07/test/Spago/Build/Monorepo.purs#L117-L120